src/NServiceBus.Core/Features/Feature.cs

```diff
@@ -64,7 +64,7 @@ protected void Defaults(Action<SettingsHolder> settings)
         /// <summary>
         ///     Called when the features is activated.
         /// </summary>
-        protected internal abstract void Setup(FeatureConfigurationContext context);
+        protected internal abstract IReadOnlyCollection<FeatureStartupTask> Setup(FeatureConfigurationContext context);
```

Comment:
Over the context works as well. In that spike I wanted to explore a more functional concept where instead of internally register (in therefore intransparently) the tasks will be returned visibly over the return method of the setup call. But I'm not hung up on that. Since pipeline and stuff currently also goes over the `FeatureConfigurationContext` it is probably more consistent to do it over the context as well.